davinci/src/main/kotlin/com/pingidentity/davinci/collector/Collectors.kt

```diff
@@ -20,9 +21,9 @@ import kotlinx.serialization.json.put
 internal fun Collectors.eventType(): String? {
     forEach {
         when (it) {
-            is SubmitCollector, is FlowCollector -> {
-                if ((it as SingleValueCollector).value.isNotEmpty()) {
-                    return it.value
+            is Submittable -> {
+                it.payload().let {
+                    return "submit"
```

Comment:
@witrisna why "return "submit"" and not the `payload` it self?

Reply:
This is for the eventType, the format that the server accept:
```json
{
  "id": "26zaz4xw9s",
  "eventName": "continue",
  "parameters": {
    "eventType": "submit",
    "data": {
      "formData": {
        "device-registration": "EMAIL"
      }
    }
  }
}
```

If there is payload of the Submittable Collector (Submit, Flow, DeviceRegistration, DeviceAuthentication), we inject "eventType": "submit"